src/Common/src/Common/Lifecycle/DefaultLifecycleProcessor.cs

```diff
@@ -221,7 +229,7 @@ public Task Stop()
 
             private async Task DoStart(ILifecycle bean)
             {
-                if (bean != null && bean != this && !bean.IsRunning && (!_autoStartupOnly || !(bean is ISmartLifecycle) || ((ISmartLifecycle)bean).IsAutoStartup))
+                if (bean != null && bean != this && !bean.IsRunning && (!_autoStartupOnly || bean is not ISmartLifecycle lifecycle || lifecycle.IsAutoStartup))
```

Comment:
I recognize it's not all new code, but should this class use `service` instead of `bean`, particularly in the exception on line 240?

Reply:
Not sure ... we are using bean all over the place, even in a few public interfaces ex. IInvocableHandlerMethod, InvocableHandlerMethod etc. So I am afraid it would be too much to rename all instances of beans to services. I did add service in parenthesis to the error message. Not sure if that would suffice ...

Reply:
My preference would be to avoid usage of `bean` unless it's directly relevant. I don't know offhand how much (if any) of the usage we currently have has actually shipped, but I know we did a round of removal/replacement before 3.0. For internal code it matters less, but I'd think logging the term `bean` at all would be more confusing than helpful cc @dtillman